plugin-maven/src/test/java/com/diffplug/spotless/maven/generic/ToggleOffOnTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2016-2020 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.maven.generic;
+
+import org.junit.Test;
+
+import com.diffplug.spotless.maven.MavenIntegrationHarness;
+
+public class ToggleOffOnTest extends MavenIntegrationHarness {
+	@Test
+	public void toggleOffOn() throws Exception {
+		writePomWithJavaSteps(
+				"<indent>",
+				"  <tabs>true</tabs>",
+				"  <spacesPerTab>1</spacesPerTab>",
+				"</indent>",
+				"<toggleOffOn />");
+		setFile("src/main/java/Main.java").toLines(
+				" Here is some stuff",
+				" No matter",
+				"//spotless:off",
+				" This won't get tabbed",
+				"//spotless:on",
+				" But this will get tabbed.");
+		mavenRunner().withArguments("spotless:apply").runNoError();
+		assertFile("src/main/java/Main.java").hasLines(
+				"	Here is some stuff",
+				"	No matter",
+				"//spotless:off",
+				" This won't get tabbed",
+				"//spotless:on",
+				"	But this will get tabbed.");
```

Comment:
It occurs to me that we don't have an equivalent test that checks `// spotless:off` and `// spotless:on` work in the Gradle plugin.

We already have one that checks `spotless:off` and `spotless:on` tags in a generic text file, but not a test with those tags inside Java comments.

What do you think about adding such a test to, say, `plugin-gradle/src/test/java/com/diffplug/gradle/spotless/ToggleOffOnTest.java`?

Reply:
I'm fine with it, but I don't think it's necessary.  I think the point of the plugin tests is to test the integration logic, not to fully exercise every feature of every step.  The only reason the tests are different is that I couldn't make an "uppercase" step in the maven plugin, so I had to make some other example.  The `spotless:off` tags work just as well in a string constant as within a comment, for better or worse (I think mostly worse).